src/plugins/discover/public/application/main/services/discover_state.ts

```diff
@@ -540,5 +541,6 @@ function createUrlGeneratorState({
     viewMode: appState.viewMode,
     hideAggregatedPreview: appState.hideAggregatedPreview,
     breakdownField: appState.breakdownField,
+    dataViewSpec: !dataView?.isPersisted() ? dataView?.toSpec() : undefined,
```

Comment:
this is the main change, using temporary data views with search sessions didn't work because the dataViewSpec wasn't provided when persisting and restoring the session